src/app/components/navigation/navigation.component.ts

```diff
@@ -0,0 +1,17 @@
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from 'src/app/services/auth.service';
+import { Router } from '@angular/router';
+@Component({
+  selector: 'app-navigation',
+  templateUrl: './navigation.component.html',
+  styleUrls: ['./navigation.component.css'],
+})
+export class NavigationComponent implements OnInit {
+  constructor(private authService:AuthService, private router:Router) {}
```

Comment:
use seprate service to fetch user details and set the user name (if logged in) in header